lms/djangoapps/courseware/courses.py

```diff
@@ -519,6 +519,34 @@ def get_course_assignment_date_blocks(course, user, request, num_return=None,
     return date_blocks
 
 
+@request_cached()
+def get_course_blocks_completion_summary(course_key, user):
```

Comment:
I'd argue for a docstring here

Reply:
So from looking at the code, I see a problem cannot be both incomplete and locked. 
1) Is that intentional? I could see an argument that locked content is _also_ incomplete content.
2) Assuming it is the desired logic, I'd say to add that information into the docstring since that is potentially something people may not expect.

Reply:
Yeah this is intentional. I'll clarify in the docstring